memote/suite/cli/runner.py

```diff
@@ -393,88 +411,290 @@ def history(model, message, rewrite, solver, location, pytest_args, deployment,
         move(new_location, os.getcwd())
     repo.git.add(".")
     repo.index.commit(message)
+    LOGGER.info("Success!")
     # Checkout the original branch.
     previous.checkout()
     LOGGER.info("Done.")
 
 
-@cli.command(context_settings=CONTEXT_SETTINGS)
-@click.help_option("--help", "-h")
-@click.option("--note", default="memote-ci access", show_default=True,
-              help="A note describing a personal access token on GitHub. "
-                   "Must be unique!")
-@click.option("--github_repository", callback=callbacks.validate_repository,
-              help="The repository name on GitHub. Usually this is configured "
-                   "for you.")
-@click.option("--github_username", callback=callbacks.validate_username,
-              help="The GitHub username. Usually this is configured for you.")
-def online(note, github_repository, github_username):
-    """Upload the repository to GitHub and enable testing on Travis CI."""
-    try:
-        repo = git.Repo()
-    except git.InvalidGitRepositoryError:
-        LOGGER.critical(
-            "'memote online' requires a git repository in order to follow "
-            "the current branch's commit history.")
-        sys.exit(1)
+def _setup_gh_repo(github_repository, github_username, note):
     password = getpass("GitHub Password: ")
-    gh = Github(github_username, password)
-    user = gh.get_user()
+
+    headers = {"Accept": "application/vnd.github.v3+json",
+               "User-Agent": "Memote Query"}
+    credentials = (github_username, password)
+
+    # Authenticate user on GitHub
     try:
-        when = user.created_at.isoformat(sep=" ")
-        LOGGER.info(
-            "Logged in to user '{}' created on '{}'.".format(user.login, when))
-    except BadCredentialsException:
+        LOGGER.info("Using your credentials to authenticate you on GitHub.")
+        response = requests.get(
+            'https://api.github.com/user', auth=credentials, headers=headers
+        )
+        response.raise_for_status()
+    except HTTPError:
```

Comment:
> that should either be handled differently or not handled at all so that the command is aborted.

As far as I understand `raise_for_status()` raises an HTTPError for any response that isn't "200"? Is your issue with the logger output saying "incorrect username or password" even though it could also be any other "4xx/ 5xx" code or with the fact that I am not responding to each possible response code individually?

Reply:
I think I got what you meant here. I'm now distinguishing between 401, other errors,and success.

Reply:
Sorry for being too short. Yes, I meant only displaying the message about incorrect username/password when the status code is 401 and otherwise the message of the error.